numpy/core/src/npysort/timsort.cpp

```diff
@@ -109,23 +106,21 @@ static inline int
 resize_buffer_(buffer_<Tag> *buffer, npy_intp new_size)
 {
     using type = typename Tag::type;
+
+    type *new_pw;
+
     if (new_size <= buffer->size) {
         return 0;
     }
 
-    if (NPY_UNLIKELY(buffer->pw == NULL)) {
-        buffer->pw = (type *)malloc(new_size * sizeof(type));
-    }
-    else {
-        buffer->pw = (type *)realloc(buffer->pw, new_size * sizeof(type));
-    }
-
+    new_pw = (type *)realloc(buffer->pw, new_size * sizeof(type));
```

Comment:
Would also be a bit nicer to just put the `type *new_pw` defnition here (or at the very top, but here is nicer).

Anyway, just some style nits, might just apply or merge later.